### PR TITLE
`SideNav` - "How to use" documentation update

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This PR updates the “how to use” documentation for the `SideNav` component.

I know it will not be simple to review this PR, there's **a lot** of content, so maybe my suggestion is to focus on the overall structure of the content, and the technical information in it: as a consumer (developer) is it clear what I am explaining? if you had to implement this component, would you have enough information to understand _broadly_ how to use it?

Note: as previously discussed with @heatherlarsen, for consistency with other components' documentation, we will keep the content - even if extremely long - in the "how to use" section of the "code" tab. In future we will consider a review of the information architecture of all the pages in a systematic way, if needed.

👉 👉 👉 **Preview:**
https://hds-website-git-sidenav-how-to-documentation-update-hashicorp.vercel.app/components/side-nav?tab=code

### :link: External links

Jira ticket: https://hashicorp.atlassian.net/browse/HDS-1826

***

### 👀 Reviewer's checklist:

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by MelSumner — APPROVED** [review]

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 1]
```diff
@@ -0,0 +1,553 @@
+
+This section provides in-depth instructions on how consumers can use the **full-featured `Hds::SideNav`** component to build a "standard" sidebar navigation with responsive behaviour, animations/transitions, support for portals, etc.
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:2]: English vs. American spelling
```suggestion
This section provides in-depth instructions on how consumers can use the **full-featured `Hds::SideNav`** component to build a "standard" sidebar navigation with responsive behavior, animations/transitions, support for portals, etc.
```

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+
+It also provides generic guidance on how to use the **layout-only `Hds::SideNav::Base`** component to build a customised sidebar navigation (if that would be necessary).
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:4]: English vs. American spelling
```suggestion
It also provides generic guidance on how to use the **layout-only `Hds::SideNav::Base`** component to build a customized sidebar navigation (if that would be necessary).
```

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+
+Given the complexity that and level of customization that an application's navigation may require, it is not possible to cover all the possible use cases in this documentation. For this reason, if you need to implement a navigation element using this component, we suggest to contact the Design Systems Team for [guidance and support](/about/support).
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:6]:
```suggestion
Given the complexity and level of customization that an application's navigation may require, it is not possible to cover all the possible use cases in this documentation. For this reason, if you need to implement a navigation element using this component, we suggest contacting the Design Systems Team for [guidance and support](/about/support).
```

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+  <:footer>
+    <Doc::Placeholder @height="36px" @text="Org/Project Select (“context switcher”)" @background="#e4e4e4" />
+  </:footer>
+</Hds::SideNav>
+
+{{!--
+this "portal" can be declared in any part of the application, and its content
+will be injected automatically in the "target" portal declared above;
+if multiple portals are declared, multiple "panels" will be rendered
+based on the nesting of the page route within the application global routing
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:316]:
```suggestion
based on the nesting of the page route within the application’s global routing
```

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+When the SideNav is used in conjunction with portals, the nesting of navigation/subnavigation levels has to match one-to-one the hierarchy of the routing, otherwise it will not work as one would expect.
+
+!!!
+
+
+#### Footer (`<:footer>`)
+
+This area usually contains the "context switchers" (eg. organization or project selectors), but technically it can contain anything (it depends on the context/application where the SideNav is used).
+
+If you want (and you probably do) that the content automatically fades in/out when the SideNav changes its "minimization" state, you have to apply the specific class `hds-side-nav-hide-when-minimized` to the top-level elements of your content.
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:357]:
```suggestion
If you want (and you probably do) the content automatically fades in/out when the SideNav changes its "minimization" state, you have to apply the specific class `hds-side-nav-hide-when-minimized` to the top-level elements of your content.
```

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+
+- when the **viewport is `desktop`** the sidebar navigation is static and has a fixed width (the width at which the viewport is considered "desktop" is controlled by a dedicated CSS variable, see below for more details)
+- when the **viewport is `mobile`** the sidebar navigation is responsive and the user can minimize or maximize its width via a toggle button (the component will take care automatically of the transitions between these two states)
+    - in this case the SideNav occupies only a minimum width in terms of page layout, even when expanded (it partially covers the main content)
+    - a toggle button is added to the SideNav, used to expand/minimize its width
+    - an overlay is automatically added to the main content area so that the user can't interact with it while the SideNav is opened (if the user clicks on the overlay, the SideNav closes and returns to its minimized state; the same happens if the user presses the `esc` key)
+
+This "responsive" behaviour can be programmatically turned off passing a value `false` to the argument `@isResponsive` in the `Hds::SideNav` component.
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:403]:
```suggestion
This "responsive" behavior can be programmatically turned off passing a value `false` to the argument `@isResponsive` in the `Hds::SideNav` component.
```

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 7]
```diff
@@ -0,0 +1,553 @@
...
+- `minimized → maximized` transition: the content appears with a fade-in effect, when the width animation is already completed (the width is maximised)
+- `maximized → minimized` transition: the content disappears at once with no transition, before the width animation starts
+
+Another option is to use the **`isMinimized` parameter**, which is useful in those cases where the content is so custom/specialized that can't just be faded in/out but needs to have a different kind of transition (eg. remain visible but change layout or respond to the width of the container). This value is passed down by the `<:header/body/footer>` named blocks as parameters, and can be used to build custom logic on the consumers' side.
+
+See the code snippet for the `<:footer>` above for examples of both approaches.
+
+#### Advanced customization
+
+Some aspects of the responsiveness/animation/transition of the SideNav are parameterized in code via CSS custom properties. It means that _in theory_ they could be customised/overwritten. This though is **something that we don't recommend**.
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:441]:
```suggestion
Some aspects of the responsiveness/animation/transition of the SideNav are parameterized in code via CSS custom properties. It means that _in theory_ they could be customized/overwritten. This though is **something that we don't recommend**.
```

File: website/docs/components/side-nav/partials/code/how-to-use.md [hunk 8]
```diff
@@ -0,0 +1,553 @@
...
+
+Navigation is a cardinal element of the UI of an application, and its standardization across the different HashiCorp products is an important end goal. Also, the implementation of this component has required considerable efforts and investments, and we should preserve it as much as possible as a unifying element across products.
+
+If you find yourself in the situation of wanting/needing to customize or change the aspect or the behaviour of the SideNav component, contact the Design Systems Team for [guidance](/about/support).
```
Comment (alex-ju) [inline website/docs/components/side-nav/partials/code/how-to-use.md:445]:
```suggestion
If you find yourself in the situation of wanting/needing to customize or change the aspect or the behavior of the SideNav component, contact the Design Systems Team for [guidance](/about/support).
```

**Review by alex-ju — COMMENTED** [review]
Took me a while to get through this one! 🤯 

Looks good overall! Left a few comments where I spotted typos and such.

Is there something we can do to prevent the portal example from overlapping the page header?

Screenshot 2023-05-18 at 21 12 22

**Comment by didoo** [comment]
> Is there something we can do to prevent the portal example from overlapping the page header?

@alex-ju Ah! I left one out 🤦‍♂️ Nice catch (fixed now).

**Review by alex-ju — APPROVED** [review]
🚀

**Pull request merged**